### ISSUE TITLE
`doc_paragraphs_missing_punctuation` fires when embedding linked images

### ISSUE THREAD

**Issue by zacknewman** [rust-lang/rust-clippy#16439]
### Summary

Not sure how this should be "fixed"; but when one follows a similar template as [`syn`](https://docs.rs/syn/latest/syn/) where the first line of the documentation is just clickable images (e.g., a GitHub logo that links to the repo), this lint fires. Appending a period isn't an option since you only want to render the images.

I actually want to use this lint since my documentation is all text and examples sans the first line of the crate documentation. Unfortunately since the lint fires on the crate documentation, the only way to silence it is at a global level forcing me to live without the lint or `deny` the lint individually which is too much of a PITA.

### Lint Name

doc_paragraphs_missing_punctuation

### Reproducer

I tried this code:

```rust
//! [![git]](https://git.philomathiclife.com/priv_sep/log.html)&ensp;[![crates-io]](https://crates.io/crates/priv_sep)&ensp;[![docs-rs]](crate)
//!
//! [git]: https://git.philomathiclife.com/git_badge.svg
//! [crates-io]: https://img.shields.io/badge/crates.io-fc8d62?style=for-the-badge&labelColor=555555&logo=rust
//! [docs-rs]: https://img.shields.io/badge/docs.rs-66c2a5?style=for-the-badge&labelColor=555555&logo=docs.rs
```

I saw this happen:

```
    Checking priv_sep v3.0.0-alpha.4.0 (/home/zack/projects/priv_sep)
error: doc paragraphs should end with a terminal punctuation mark
 --> src/lib.rs:1:144
  |
1 | ...v_sep)&ensp;[![docs-rs]](crate)
  |                                   ^ help: end the paragraph with some punctuation: `.`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.93.0/index.html#doc_paragraphs_missing_punctuation
  = note: `-D clippy::doc-paragraphs-missing-punctuation` implied by `-D clippy::restriction`
  = help: to override `-D clippy::restriction` add `#[allow(clippy::doc_paragraphs_missing_punctuation)]`

error: could not compile `priv_sep` (lib) due to 1 previous error
```

I expected to see this happen:

No lints to fire.

### Version

```text
rustc 1.93.0 (254b59607 2026-01-19)
binary: rustc
commit-hash: 254b59607d4417e9dffbc307138ae5c86280fe4c
commit-date: 2026-01-19
host: x86_64-unknown-linux-gnu
release: 1.93.0
LLVM version: 21.1.8
```

**Comment by Francisco-Andre-Martins** [comment]
Can reproduce locally @rustbot claim
